concepts/eventing/performance evaluation/nats-jetstream/different-cloud-providers/nats-throughput-test.md

```diff
@@ -0,0 +1,111 @@
+# Throughput Evalation of NATS Jetstream on different Cloud Providers
+
+## Table of Contents
+- Test Setup
+- Test Results
+- How tests are executed
+
+## Test Setup
+* Kyma CLI version: `2.6.2`
+* Kyma: 
+  * Version: main 
+  * Production Profile
+  * JetStream with File Storage
+  * NATS JetStream Version: `2.9.0` 
+* K8s cluster:
+  * Kubernetes v1.24.6
+  * Gardener cluster [Nodes: 2(min) to 6(max)]
+  * Machine Type: `cpu: 2, memory: 16GB`
+
+* Kyma deploy command:
+    ```
+    kyma deploy --source=main --profile production ...
+    ```
+* Testing tool: [NATS Benchmarking Tool](https://docs.nats.io/using-nats/nats-tools/nats_cli/natsbench)
+<br/><br/>
+
+## Test Results
+As you see from the table all the NATS Jetstream throughput tests were similar on GCP, Azure, and AWS. 
+
+| Cloud Provider | msgs/sec (NATS Bench) | msgs | storage type | pub/sub | CPU / Memory |
+| --- | --- | --- | --- | --- | --- |
+| Azure | 4130 - 4313 | 10000 | SSD | 5/5 | 4 / 16 |
+| GCP | 2000 - 2500 | 10000 | SSD | 5/5 | 4 / 15 |
+| AWS | 3441 - 4289 | 10000 | SSD | 5/5 | 4 / 16 |
+|  |  |  |  |  |  |
+| Azure | 1630 - 2682 | 10000 | SSD | 1/5 | 4 / 16 |
+| GCP | 850  - 1320 | 10000 | SSD | 1/5 | 4 / 15 |
+| AWS | 1739 - 2071 | 10000 | SSD | 1/5 | 4 / 16 |
+|  |  |  |  |  |  |
+| GCP | 2297-2447 | 10000 | HDD | 5/5 | 4 / 16 |
+|  |  |  |  |  |  |
+| Azure | 940   - 1198 | 10000 | SSD | 5/5 | 2 / 16 |
+| Azure | 502  - 584 | 10000 | SSD | 1/5 | 2 / 16 |
+<br/>
+
+## How tests are executed
+Run the following commands against the installed NATS Jetstream
+
+### Stop EC
+
+```
+$ kubectl -n kyma-system scale --replicas 0 deploy eventing-controller
+```
+
+### Get into nats-box pod to run benchmarks in k8s cluster:
+```
+$ kubectl -n kyma-system run -i --rm --tty nats-box --image=natsio/nats-box --restart=Never
+```
+### Create consumer
+Create a durable consumer simulating Kyma NATS Jetstream consumer:
+```
+$ nats con add sap test-consumer --server=nats://eventing-nats:4222" --deliver-group test-consumer-GROUP --target test-consumer-DELIVERY --filter "kyma.sap.kyma.custom.natsbenchmark.execution.created.v1" --deliver new --ack explicit --replay instant --max-deliver 100 --max-pending 10 --flow-control --heartbeat 1m --wait 30s
```

Comment:
Curios how `--server=nats://eventing-nats:4222"` works under the hood to discover NATS server without specifying in which namespace NATS is deployed?

Reply:
exactly, as the oats-box container is started inside `Kyma-system`namespace.